config/interpolate_funcs.go

```diff
@@ -640,6 +641,44 @@ func appendIfMissing(slice []string, element string) []string {
 	return append(slice, element)
 }
 
+// interpolationFuncFilter implements the "filter" function that
+// removes elements from a list
+func interpolationFuncFilter() ast.Function {
+	return ast.Function{
+		ArgTypes:   []ast.Type{ast.TypeList, ast.TypeList, ast.TypeString},
+		ReturnType: ast.TypeList,
+		Callback: func(args []interface{}) (interface{}, error) {
+			var list []string
+
+			argument, _ := args[0].([]ast.Variable)
+			keys, _ := args[1].([]ast.Variable)
+			searchkey, _ := args[2].(string)
+
+			if len(argument) != len(keys) {
+				return nil, fmt.Errorf("length of lists should be equal")
+			}
+
+			for i, element := range argument {
+				if element.Type != ast.TypeString {
+					return nil, fmt.Errorf(
+						"only works for flat lists, this list contains elements of %s",
+						element.Type.Printable())
+				}
+				if keys[i].Type != ast.TypeString {
+					return nil, fmt.Errorf(
+						"only works for flat lists, this list contains elements of %s",
+						keys[i].Type.Printable())
+				}
+				if keys[i].Value.(string) == searchkey {
+					list = append(list, element.Value.(string))
+				}
+			}
+
+			return stringSliceToVariableValue(list), nil
```

Comment:
Was it an intention here to just return an empty list? (genuine question - not suggesting it should be changed!)
If it was - that's fine - but I think we should document that in the docs, e.g. `No match will result in empty list` - just to avoid questions whether or not it will error out in such cases.